docs/src/description/convenience-syntax.md

```diff
@@ -0,0 +1,55 @@
+# Convenience syntax
+
+To make writing constraints easier, AirScript provides a number of syntactic conveniences. These are described in this section.
+
+## List Comprehension
+
+List comprehension provides syntactic convenience for creating new vectors. It is similar to the list comprehension syntax in Python. The following examples show how to use List Comprehension in AirScript.
+
+```
+let x = [a * 2 for a in b]
+```
+This will create a new vector with the same length as `b` and each element will be twice the corresponding element in `b`.
+
+```
+let x = [a + b for (a, b) in (c, d)]
+```
+This will create a new vector with the same length as `c` and `d` and each element will be the sum of the corresponding elements in `c` and `d`. This will throw an error if `c` and `d` vectors are of unequal lengths.
+
+```
+let x = [2^i * a for (i, a) in (0..5, b)]
+```
+Ranges can also be used as iterables. This will create a new vector with length 5 and each element will be the corresponding element in `b` multiplied by 2 raised to the power of the index of the element. This will throw an error if `b` is not of length 5.
+
+```
+let x = [m + n + o for (m, n, o) in (a, 0..5, c[0..5])]
+```
+Slices can also be used as iterables. This will create a new vector with length 5 and each element will be the sum of the corresponding elements in `a`, `b` and `c`. This will throw an error if `a` is not of length 5 or if c is of length less than 5.
+
+## List Folding
+
+List folding provides syntactic convenience for folding vectors into expressions. It is similar to the list folding syntax in Python. List folding can be applied on vectors, list comprehension or identifiers referring to vectors and list comprehension. The following examples show how to use List Folding in AirScript.
```

Comment:
nit: 
- this should be "...applied to vectors..." instead of "applied on"
- let's uncapitalize "list folding" in the last sentence